clarify (#3)
ERC4626-Alliance/EIPs: EIPS/eip-x.md

The following standard extends [ERC-4626](./eip-4626.md) by adding support for asynchronous deposit and redemption flows. The async flows are called "Requests".

> New methods are added to asynchronously request a deposit or redeem call, and view the pending status of the request. The existing deposit, mint, withdraw, and redeem ERC-4626 methods are used for executing claimable Requests. 

Reviewer: nit: `call` is a term we're not using anywhere else, wouldn't introduce that here
```suggestion
New methods are added to asynchronously request a deposit or redemption, and view the pending status of the request. The existing deposit, mint, withdraw, and redeem ERC-4626 methods are used for executing claimable Requests. 
```

Reviewer: Also even more nit: we should align on `request` vs `Request` (and `vault` vs `Vault`)

Reviewer 2: Request and Vault make sense!

Author: +1 Request and Vault we can do one pass for that at the end

> Implementations can choose to whether to add asynchronous flows for deposit,  redeem, or both. 

Reviewer: ```suggestion
Implementations can choose to whether to add asynchronous flows for deposits, redemptions, or both. 
```

### Definitions:
The existing definitions from [ERC-4626](./eip-4626.mn) apply. In addition, this spec defines:
- request: a function call that initiates an asynchronous deposit/redemption flow
- pending request: the state where a request has been made but is not yet claimable
- claimable: the Vault's step of processing the request and enabling the user to claim corresponding shares (for async deposit) or assets (for async redeem)
- claim: the corresponding Vault method to complete a request (e.g. `deposit` claims `shares` from `requestDeposit`)

> - operator: the ethereum account specified by the requester which has has the right to claim a given request once it is claimable

Reviewer: `requester` is not a term we otherwise use. And I think `ethereum account` is implied
```suggestion
- operator: the account specified by the sender of the request which has has the right to claim a given request once it is claimable
```

| Claimable |                                  | <i>Internal request fulfillment</i><br>pendingDepositRequest[msg.sender] -= assets<br>maxDeposit[operator] += assets<br> |
|   Claimed | deposit(assets, receiver)        |                                                  maxDeposit[msg.sender] -= assets<br>vault.balanceOf[receiver] += shares |

> An important vault invariant is that following a request(s), the cumulative requested quantity MUST equal `pendingDepositRequest + maxDeposit - claimed`. The previous invariant assumes there are no cancel requests, whose behavior is discussed below.

Reviewer 2: if requests are cancelled, wouldn't `maxDeposit` go up the same amount as `pendingDepositRequest` goes down? invariant of `pendingDepositRequest + maxDeposit` should still hold right. might be missing something with `claimed`

Author: It should still hold as an inequality but pendingDeposit may go down before maxDeposit recovers

> An important vault invariant is that following a request(s), the cumulative requested quantity MUST equal `pendingDepositRequest + maxDeposit - claimed`. The previous invariant assumes there are no cancel requests, whose behavior is discussed below.

Reviewer: ```suggestion
An important vault inequality is that following a request(s), the cumulative requested quantity MUST be more than `pendingDepositRequest + maxDeposit - claimed`. The previous inequality assumes there are no cancel requests, whose behavior is discussed below.
```

| Claimable |                                  | <i>Internal request fulfillment</i><br>pendingDepositRequest[msg.sender] -= assets<br>maxDeposit[operator] += assets<br> |
|   Claimed | deposit(assets, receiver)        |                                                  maxDeposit[msg.sender] -= assets<br>vault.balanceOf[receiver] += shares |

> An important vault inequality is that following a request(s), the cumulative requested quantity MUST be more than `pendingDepositRequest + maxDeposit - claimed`. The previous inequality assumes there are no cancel requests, whose behavior is discussed below.

Author: ```suggestion
An important vault inequality is that following a request(s), the cumulative requested quantity MUST be more than `pendingDepositRequest + maxDeposit - claimed`. The sources of inequality are fees and cancellation requests, otherwise this would be a strict equality.
```

The methods `pendingDepositRequest` and `pendingRedeemRequest` are estimates useful for display purposes, and can be outdated due to the asynchronicity.

> In general, asynchronicity concerns make state transitions in the vault much more complex and vulnerable to security risks. Access control on vault operations, clear documentation of state transitioning and invariant checks should all be performed to mitigate these risks.

Reviewer: ```suggestion
In general, asynchronicity concerns make state transitions in the vault much more complex and vulnerable to security risks. Access control on vault operations, clear documentation of state transitioning, and invariant checks should all be performed to mitigate these risks.
```
